ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivity.kt

```diff
@@ -38,10 +42,26 @@ internal class PaywallActivity : ComponentActivity(), PaywallViewListener {
         }
     }
 
+    private fun getFontProvider(): FontProvider? {
+        return getArgs()?.fonts?.let { fonts ->
+            val fontsMap = fonts.mapValues { entry ->
+                entry.value?.let { fontRes ->
+                    ResourcesCompat.getFont(this, fontRes)?.let { FontFamily(it) }
+                }
+            }
+            object : FontProvider {
+                override fun getFont(type: TypographyType): FontFamily? {
+                    return fontsMap[type]
+                }
+            }
+        }
+    }
```

Comment:
Feel free to ignore but I find this function a bit hard to read due to the fact that the return is lifted and it creates a FontProvider at the end.

Maybe extracting first the `fontsMap` then returning the FontProvider or null would make it easier to read. If you think it's fine, don't change it, just felt I should mention it.